### PR TITLE
Fix recurrent docs (inputs -> sequences)

### PR DESCRIPTION
**Pull request by rizar**
@udibr , anything else?

### CODE REVIEW

File: blocks/bricks/recurrent.py [hunk 1]
```diff
@@ -59,27 +59,24 @@ def initial_state(self, state_name, batch_size, *args, **kwargs):
...
-    This decorator allows you to use implementation of an RNN
-    transition to process sequences without writing the
-    iteration-related code again and again. In the most general form
-    information flow of a recurrent network can be described as
-    follows: depending on the context variables and driven by input
-    sequences the RNN updates its states and produces output sequences.
-    Thus the input variables of your transition function play one of
-    three roles: an input, a context or a state. These roles should be
-    specified in the method's signature to make iteration possible.
+    This decorator allows you to use implementation of an RNN transition to
```
Comment (bartvm) [inline blocks/bricks/recurrent.py]: I'm not following this sentence. Maybe "This decorator allows you to implement a recurrent transition to process sequences..."?

File: blocks/bricks/recurrent.py [hunk 2]
```diff
@@ -59,27 +59,24 @@ def initial_state(self, state_name, batch_size, *args, **kwargs):
...
+    process sequences without writing the iteration-related code again and
+    again. In the most general form information flow of a recurrent network
```
Comment (bartvm) [inline blocks/bricks/recurrent.py]: In *its* most general form *the* information...

File: blocks/bricks/recurrent.py (hunk 1 above)
Comment (bartvm) [inline blocks/bricks/recurrent.py]: Also, *the* implementation

File: blocks/bricks/recurrent.py [hunk 3]
```diff
@@ -59,27 +59,24 @@ def initial_state(self, state_name, batch_size, *args, **kwargs):
...
     ----------
-    inputs : list of strs
-        Names of the arguments of the apply method that play input
-        roles.
+    sequences : list of strs
+        Specifies which of the arguments are elements of input sequences.
     states : list of strs
-        Names of the arguments of the apply method that play state
-        roles.
+        Specifies which of the arguments that are states.
```
Comment (bartvm) [inline blocks/bricks/recurrent.py]: ~~that~~

File: blocks/bricks/recurrent.py (hunk 1 above)
Comment (rizar) [inline blocks/bricks/recurrent.py]: All right, although I meant only to fix a blatant argument name error in this PR,  I will also rewrite this chuck in a more readable way. Thanks for corrections!

Comment (bartvm) [inline blocks/bricks/recurrent.py]: Oops, I didn't realise the rest was just a different wrapping of the same text.

**Comment by rizar** [comment]
I rewrote the `recurrent` docs, which does not make them perfect, but I think somewhat more understandable.

**Comment by rizar** [comment]
@bartvm what is the correct way to reference tutorials from docstrings?

**Comment by bartvm** [comment]
Something like this should work:
```
Please read :doc:`the tutorial on RNNS <rnn>` for more information.
```

**Comment by udibr** [comment]
If I remeber correctly the tutorial also needs a re-write

**Comment by rizar** [comment]
@udibr , @dmitriy-serdyuk , @bartvm , can somebody of you guys give a read to these updated docs?

**Comment by bartvm** [comment]
LGTM, we'll probably want examples (doctests) at one point, but this is better than it was.

**Pull request merged**